src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,299 @@
+            {renderEntries()}
+        </RootContainer>
+    )
+}
+
+interface IRootContainerProps extends MainInterface{
+    width?:number
+}
+
+const RootContainer=styled.div<IRootContainerProps>`
+border:2px solid black;
+${({width,...props}):string=>`
+${width?`width:${width}px;`:'width:fit-content;'}
+${Main({...props})}
+`}
+`
+
+interface IInfo{
+    setDelayedLabel:React.Dispatch<React.SetStateAction<string | undefined>>;
+}
+
+interface ICommonEntryProps{
+    fontSize?:string;
+    bold?:boolean;
+    separatorWidth?:number;
+    editMode?:boolean;
+    margin?:string;
+    padding?:string;
+}
+
+interface IHeadingEntryProps extends ICommonEntryProps{
+    label:string;
+    justifyContent?:string;
+    secondLabel?:string;
+    infoRef?:React.MutableRefObject<IInfo | null>;
+    editable?:boolean;
+}
+
+const HeadingEntry=forwardRef<HTMLDivElement,IHeadingEntryProps>(({label,editMode,editable,separatorWidth=1,secondLabel,infoRef,...props},ref):React.ReactElement=>{
+    const [delayedLabel,setDelayedLabel]=useState<string>()
+    const [secondLabelState,setSecondLabelState]=useState(secondLabel)
+
+    /**
+     * this updates info in the parent component
+     */
+    useLayoutEffect(()=>{
+        if(infoRef) infoRef.current={setDelayedLabel}
+    },[])
+
+    /**
+     * updates state for second label based on event fired
+     * @param event {React.ChangeEvent<HTMLInputElement>} - the fired event 
+     */
+    const updateSecondLabel=useCallback(({target}:React.ChangeEvent<HTMLInputElement>)=>{
+        setSecondLabelState(target.value)
+    },[]) 
+
+    /**
+     * renders the second label if exists
+     * @returns {JSX.Element | null} the rendered second label, in case that exists, depending on edit mode and if the entry is 
+     * editable
+     */
+    const renderSecondLabel=useCallback(()=>{
+        if(secondLabelState){
+            if(editMode&&editable)
+                return <Input value={secondLabelState} onChange={updateSecondLabel} width={HEADER_ENTRY_INPUT_WIDTH} />
+            return <div>{secondLabelState}</div>
+        }
+        return null
+    },[secondLabelState])
+
+    return (
+        <EntryContainer separatorWidth={separatorWidth} {...props} ref={ref}>
+            <div>{label||delayedLabel}</div>
+            {renderSecondLabel()}
+        </EntryContainer>
+    )
+})
+
+interface IEntryProps extends ICommonEntryProps{
+    amount?:number;
+    dailyAmount:number;
+    label:string;
+    unity:string;
+    indentationNumber?:number;
+    indentationSize?:number;
+}
+
+const Entry:React.FC<IEntryProps>=({
+    amount=0,
+    dailyAmount,
+    label,
+    unity,
+    editMode,
+    bold=false,
+    separatorWidth=1,
+    indentationNumber=0,
+    indentationSize=4,
+    ...props
+}):React.ReactElement=>{
+    const [amountState,setAmountState]=useState<string>(amount.toString())
+    
+    /**
+     * this computes the daily value percentage
+     */
+    const dailyValuePercentage= useMemo(()=>{
```

Comment:
missing TypeScript Return types for function

const Example Function  = (): void => {}

Reply:
first of all, it's not a function, it's a value. and second, its typed is implicit. 
![Captura de pantalla 2021-03-30 a las 13 56 36](https://user-images.githubusercontent.com/29568460/112984927-cf410e80-915f-11eb-8cd8-a1394cc0189c.png)